src/Sylius/Bundle/ApiBundle/Controller/GetStatisticsAction.php

```diff
@@ -0,0 +1,194 @@
+            new \DateTimeImmutable($parameters['endDate']),
+        );
+
+        try {
+            $this->validateEndDateIsNotBeforeStartDate($period);
+            $this->validateEndDateAgainstInterval($period);
+        } catch (\InvalidArgumentException $exception) {
+            return new JsonResponse(
+                data: $this->serializer->serialize(['message' => $exception->getMessage()], 'json'),
+                status: Response::HTTP_BAD_REQUEST,
+                json: true,
+            );
+        }
+
+        $query = new GetStatistics($period, $parameters['channelCode']);
+
+        try {
+            $result = $this->handle($query);
+        } catch (HandlerFailedException $exception) {
+            return new JsonResponse(
+                data: $this->serializer->serialize(['message' => $exception->getMessage()], 'json'),
+                status: Response::HTTP_NOT_FOUND,
+                json: true,
+            );
+        }
+
+        return new JsonResponse(data: $this->serializer->serialize($result, 'json'), json: true);
+    }
+
+    private function validateRequiredParameters(Request $request): void
+    {
+        foreach ($this->requiredParameters as $parameterName => $parameterType) {
+            $parameter = $request->query->get($parameterName);
+
+            if ($parameter === null) {
+                $this->addViolation($parameterName, sprintf('Parameter "%s" is required.', $parameterName));
+
+                continue;
+            }
+
+            if (empty($parameter)) {
+                $this->addViolation($parameterName, sprintf('Parameter "%s" cannot be empty.', $parameterName));
+
+                continue;
+            }
+
+            if ($parameterType === 'dateTime' && !$this->isISO8601DateTimeWithNoTimezone($parameter)) {
+                $this->addViolation($parameterName, sprintf(
+                    'Parameter "%s" must be a valid ISO8601 date time string without timezone.',
+                    $parameterName,
+                ));
+            } elseif ($parameterType === 'dateInterval' && !$this->isValidInterval($parameter)) {
+                $this->addViolation($parameterName, sprintf(
+                    'Parameter "%s" must be a valid DateInterval string.',
+                    $parameterName,
+                ));
+            } elseif ($parameterType === 'int' && filter_var($parameter, \FILTER_VALIDATE_INT) === false) {
+                $this->addViolation($parameterName, sprintf(
+                    'Parameter "%s" must be an integer.',
+                    $parameterName,
+                ));
+            } elseif (!is_string($parameter) || is_numeric($parameter)) {
+                $this->addViolation($parameterName, sprintf(
+                    'Parameter "%s" must be a string.',
+                    $parameterName,
+                ));
+            }
+        }
+    }
+
+    private function isISO8601DateTimeWithNoTimezone(?string $dateTime = null): bool
```

Comment:
While manual validators are alright in some cases, this feels complex enough to warrant use of validation library.

I see three options,
1. Create `GetStatisticsRequest` class, which can be validated, like it was done for SyliuShopApiPlugin with regular SF validator
    - https://github.com/Sylius/ShopApiPlugin/blob/1.7/src/Request/Cart/PutSimpleItemToCartRequest.php
    - https://github.com/Sylius/ShopApiPlugin/blob/1.7/src/Resources/config/validation/cart/PutSimpleItemToCartRequest.xml
2. Symfony validator has `callable` functionality, which I think can be used in this case directly in controller without any intermediate DTOs or configuration files (Didn't try myself) https://symfony.com/doc/5.4/validation.html#validation-callables
3. Use other validation libs, like respect https://respect-validation.readthedocs.io/en/latest/feature-guide/

![image](https://github.com/Sylius/Sylius/assets/870747/3647ef19-a6d5-46ab-ab18-8061439ac882)